### PR TITLE
feat: 7702 safe

### PR DESCRIPTION
**Pull request by chris13524**
WIP: attempting to use 7702 with a Safe

Possibly depends on: https://github.com/WalletConnect/alto/pull/1/files

### CODE REVIEW

**Comment by jackpooleywc** [comment]
> WIP: attempting to use 7702 with a Safe
> 
> Possibly depends on: https://github.com/WalletConnect/alto/pull/1/files

We don't want to upstream this I assume so maybe we should keep this in a branch rather than merging?

File: test/scripts/forked_state/docker-compose.yaml [hunk 1]
```diff
@@ -1,21 +1,28 @@
 services:
   anvil:
```
Comment (jackpooleywc) [inline test/scripts/forked_state/docker-compose.yaml:2]: this should all go into `test/scripts/7702`

File: crates/yttrium/src/bundler/client.rs [hunk 2]
```diff
@@ -298,6 +298,7 @@ mod tests {
                 max_priority_fee_per_gas: max_priority_fee_per_gas.into(),
                 paymaster: paymaster,
                 paymaster_data: paymaster_data,
+                // authorization_list: None,
```
Comment (jackpooleywc) [inline crates/yttrium/src/bundler/client.rs:301]: I'm assuming this is going to be used in future PRs?

**Review by jackpooleywc — COMMENTED** [review]
Looks good, just need to move the `mock-aa-environment` fork changes to the `test/scripts/7702/docker-compose.yaml`

**Comment by chris13524** [comment]
> > Possibly depends on: https://github.com/WalletConnect/alto/pull/1/files
> 
> We don't want to upstream this I assume so maybe we should keep this in a branch rather than merging?

I don't think dependency forks have any affect on merging this PR

File: crates/yttrium/src/bundler/client.rs (hunk 2 above)
Comment (chris13524) [reply crates/yttrium/src/bundler/client.rs:301]: Yeah possibly

File: test/scripts/forked_state/docker-compose.yaml (hunk 1 above)
Comment (chris13524) [reply test/scripts/forked_state/docker-compose.yaml:2]: Why? The 7702 infra is compatible with the not. Separating these will make testing more complicated as you have to run only specific tests against one or the other environment vs just running all available

**Comment by jackpooleywc** [comment]
> > > Possibly depends on: https://github.com/WalletConnect/alto/pull/1/files
> > 
> > 
> > We don't want to upstream this I assume so maybe we should keep this in a branch rather than merging?
> 
> I don't think dependency forks have any affect on merging this PR

No I meant we might not want to merge: https://github.com/WalletConnect/alto/pull/1

**Comment by chris13524** [comment]
> No I meant we might not want to merge: https://github.com/WalletConnect/alto/pull/1No I meant we might not want to merge: https://github.com/WalletConnect/alto/pull/1


Ah, yes. We can just leave in the branch

**Review by jackpooleywc — DISMISSED** [review]

**Pull request merged**